Belindas-closet-nextjs-4-34-add-product-page-link

taylorpapke opened the pull request:
resolves: https://github.com/SeattleColleges/belindas-closet-nextjs/issues/33

Robel-003 reviewed (approved):
LGTM. Working on my end. I'm able to enter in a product and submit. Also shows the product after submission. Great job!

bcko commented on app/api/auth/product-page/page.tsx:
```diff
@@ -0,0 +1,50 @@
+  render() {
+    return (
+      <div>
+        <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
```
Let's unify CSS styling. Instead of inline style, let's use CSS module.

bcko commented on app/api/auth/product-page/page.tsx:
```diff
@@ -0,0 +1,50 @@
+
+import React, { Component } from 'react';
+
+class AddProduct extends Component {
```
the React community moved to functional components from class components.  Let's use functional components instead.

bcko replied:
Here are some resources

https://www.geeksforgeeks.org/reactjs-functional-components/
https://www.educative.io/blog/react-component-class-vs-functional
https://www.geeksforgeeks.org/differences-between-functional-components-and-class-components-in-react/

bcko commented on app/api/auth/product-page/page.tsx:
```diff
@@ -0,0 +1,50 @@
+
+  handleChange = (e: { target: { value: any; }; }) => {
+    let updatedProductName = e.target.value;
+    this.setState({ productName: updatedProductName });
```
Let's use state hooks 

https://legacy.reactjs.org/docs/hooks-state.html
https://react.dev/learn/state-a-components-memory
https://react.dev/reference/react/useState

bcko commented on app/api/auth/product-page/page.tsx:
```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+
+class AddProduct extends Component {
+  state = {
+    productName: '',
+    showText: false
```
Since we are using Typescript, you can give types to your state. Here are some examples

https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/hooks/
https://www.robinwieruch.de/typescript-react-usestate/

bcko commented:
@taylorpapke 
Since your code is working, I am fine with you merging this pull request and opening a subsequent pull request to address some code convention issues raised in this pull request.

taylorpapke commented:
Updated PR with code revisions. Realized that the feature of displaying the entered text on the screen was irrelevant given that the submission's next step is to make its way to the database. Anything that would later be displayed on our user interface would be shown based off of a request to the database through the backend and not through some process on the front end by itself. The other following changes were made:
- updated coding conventions to use a function (constant with arrow notation). 
- changes to incorporate useState hooks.
- inline css removed and css module added to the product-page file accessed through imports in the .tsx file.

Seiyaroo reviewed (approved):
LGTM
